### PR TITLE
Improve IBL code for new version of VTK

### PR DESCRIPTION
**Pull request by mwestphal**
- VTK will soon start supporting IBL without texture
- Add dedicated codepath for this
- Partially fix https://github.com/f3d-app/f3d/issues/935
- This PR will be updated once VTK changes are merged

### CODE REVIEW

File: library/VTKExtensions/Rendering/vtkF3DRenderer.cxx [hunk 1]
```diff
@@ -766,14 +775,15 @@ void vtkF3DRenderer::ConfigureHDRITexture()
     this->SetEnvironmentTexture(this->HDRITexture);
 
 #if VTK_VERSION_NUMBER >= VTK_VERSION_CHECK(9, 2, 20221220)
-    // Force modified on the spherical harmonics
-    // to avoid them updating themselves
-    // TODO add support for not providing a texture and still using IBL in VTK
-    // https://github.com/f3d-app/f3d/issues/935
+    // No cache support before that, not needed
+#if VTK_VERSION_NUMBER < VTK_VERSION_CHECK(9, 3, 20230830) // TODO date
```
Comment (Meakk) [inline library/VTKExtensions/Rendering/vtkF3DRenderer.cxx:779]: Can you merge the two ifs?

File: library/VTKExtensions/Rendering/vtkF3DRenderer.cxx [hunk 2]
```diff
@@ -751,12 +757,15 @@ void vtkF3DRenderer::ConfigureHDRITexture()
     }
     else
     {
-      // Reading the HDRI texture is actually not needed
-      // create a dummy one instead
-      // TODO add support for not providing a texture and still using IBL in VTK
-      // https://github.com/f3d-app/f3d/issues/935
+#if VTK_VERSION_NUMBER >= VTK_VERSION_CHECK(9, 3, 20230830) // TODO date
```
Comment (Meakk) [inline library/VTKExtensions/Rendering/vtkF3DRenderer.cxx:760]: > // TODO date

It's not merged in VTK yet?

Comment (mwestphal) [reply library/VTKExtensions/Rendering/vtkF3DRenderer.cxx:760]: No, I'm making myself ready to merge everything together.

File: library/VTKExtensions/Rendering/vtkF3DRenderer.cxx (hunk 1 above)
Comment (mwestphal) [reply library/VTKExtensions/Rendering/vtkF3DRenderer.cxx:779]: no, this code is need between 20221220 and 20230830

Comment (Meakk) [reply library/VTKExtensions/Rendering/vtkF3DRenderer.cxx:779]:
```cpp
#if VTK_VERSION_NUMBER >= VTK_VERSION_CHECK(9, 2, 20221220) && VTK_VERSION_NUMBER < VTK_VERSION_CHECK(9, 3, 20230830)
// ...
#endif
```

**Review by Meakk — APPROVED** [review]
Lgtm

**Pull request merged**